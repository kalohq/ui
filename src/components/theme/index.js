/* @flow */
export {default} from './theme';
