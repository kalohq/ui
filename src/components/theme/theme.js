/* @flow */

import tokens from '../../design-tokens/tokens.theme.js';

/**
 * ThemeProvider Kalo Theme
 * 
 * This is the core CSS-in-JS theme that is passed in to the
 * global ThemeProvider. It is mainly composed of our core
 * design tokens.
 */

const theme = {
  space: [0, 8, 16, 32, 64],
  breakpoints: [],
  ...tokens,
};

export default theme;
