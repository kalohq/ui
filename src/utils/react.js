import ReactDOM from 'react-dom';

/**
 * Focus a ref’d react input when it is mounted
 * @param ref
 */
export function focusOnMount(ref) {
  const el = ReactDOM.findDOMNode(ref);

  if (el) {
    // Give focus and put the cursor at the end of
    // the value.
    el.focus();
    el.value = el.value;
  }
}

/**
 * Select the text of a ref’d react input when it is mounted
 * @param ref
 */
export function selectOnMount(ref) {
  if (ref && ref.select) {
    ref.select();
  }
}
