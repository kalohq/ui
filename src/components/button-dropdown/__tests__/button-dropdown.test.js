/* @flow */
/* eslint-env jest */
import * as React from 'react';
import * as emotion from 'emotion';
import {createSerializer} from 'jest-emotion';
import renderer from 'react-test-renderer';
import {ThemeProvider} from 'emotion-theming';

import theme from 'components/theme';
import ButtonDropdown from 'components/button-dropdown';

jest.mock('react-dom', () => ({
  findDOMNode: () => {},
  unstable_renderSubtreeIntoContainer: () => {},
}));

expect.addSnapshotSerializer(createSerializer(emotion));

describe('ButtonDropdown', () => {
  const defaultProps = {
    size: 'medium',
    theme: 'tertiary',
    selectItems: [
      {
        title: 'Pending',
        onClick: () => {},
      },
      {
        title: 'This is a link',
        minWidth: 225,
        component: 'a',
        componentProps: {
          href: 'http://google.com',
          target: '_blank',
        },
      },
    ],
  };
  const create = (props = {}) =>
    renderer
      .create(
        <ThemeProvider theme={theme}>
          <ButtonDropdown {...defaultProps} {...props} />
        </ThemeProvider>
      )
      .toJSON();

  test('should render shallow component ok', () => {
    const element = create();
    expect(element).toMatchSnapshot();
  });

  test('should render with a checkbox', () => {
    const element = create({
      checkboxProps: {
        onClick: () => {},
      },
    });

    expect(element).toMatchSnapshot();
  });
});
