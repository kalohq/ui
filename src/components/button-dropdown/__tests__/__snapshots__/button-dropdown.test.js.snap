// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonDropdown should render shallow component ok 1`] = `
.emotion-1 {
  margin-left: 8px;
  margin-right: -8px;
}

.emotion-4 {
  position: absolute;
  top: calc(100% + 2px);
  min-width: calc(100% + 4px);
  left: -2px;
}

<button
  className="ui-btn ui-btn--medium ui-btn--tertiary emotion-0 emotion-7"
  data-test="ui-button"
  disabled={false}
  name={undefined}
  onClick={[Function]}
  role="button"
  style={Object {}}
  type={undefined}
>
  <div
    className="ui-btn__react-placeholder-message"
  >
    <i
      className="ui-icon ui-icon--size-20 fill-currentColor emotion-0 emotion-1"
      onClick={undefined}
      style={Object {}}
      title={undefined}
    >
      <svg
        aria-hidden="true"
        height={20}
        width={20}
      >
        <use
          xlinkHref="#keyboard_arrow_down"
        />
      </svg>
    </i>
  </div>
  <div
    className="ui-btn__react-visible-message"
    name="message"
    title={undefined}
  >
    <i
      className="ui-icon ui-icon--size-20 fill-currentColor emotion-0 emotion-1"
      onClick={undefined}
      style={Object {}}
      title={undefined}
    >
      <svg
        aria-hidden="true"
        height={20}
        width={20}
      >
        <use
          xlinkHref="#keyboard_arrow_down"
        />
      </svg>
    </i>
    <div
      className="emotion-4 emotion-5"
    >
      <div
        style={
          Object {
            "display": "none",
          }
        }
      />
    </div>
  </div>
  <div
    className="ui-btn__react-success-message"
  />
</button>
`;

exports[`ButtonDropdown should render with a checkbox 1`] = `
.emotion-3 {
  margin-left: 8px;
  margin-right: -8px;
}

.emotion-8 {
  position: absolute;
  top: calc(100% + 2px);
  min-width: calc(100% + 4px);
  left: -2px;
}

.emotion-1 {
  position: relative;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-align-content: flex-start;
  -ms-flex-line-pack: start;
  align-content: flex-start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<button
  className="ui-btn ui-btn--medium ui-btn--tertiary emotion-2 emotion-11"
  data-test="ui-button"
  disabled={false}
  name={undefined}
  onClick={[Function]}
  role="button"
  style={Object {}}
  type={undefined}
>
  <div
    className="ui-btn__react-placeholder-message"
  >
    <div
      className="ui-checkbox ui-checkbox--large emotion-0 emotion-1"
      data-test="ui-checkbox"
      name={undefined}
      style={Object {}}
    >
      <input
        checked={false}
        disabled={false}
        id="checkbox1"
        onChange={[Function]}
        type="checkbox"
      />
      <label
        htmlFor="checkbox1"
        onClick={[Function]}
      />
    </div>
    <i
      className="ui-icon ui-icon--size-20 fill-currentColor emotion-2 emotion-3"
      onClick={undefined}
      style={Object {}}
      title={undefined}
    >
      <svg
        aria-hidden="true"
        height={20}
        width={20}
      >
        <use
          xlinkHref="#keyboard_arrow_down"
        />
      </svg>
    </i>
  </div>
  <div
    className="ui-btn__react-visible-message"
    name="message"
    title={undefined}
  >
    <div
      className="ui-checkbox ui-checkbox--large emotion-0 emotion-1"
      data-test="ui-checkbox"
      name={undefined}
      style={Object {}}
    >
      <input
        checked={false}
        disabled={false}
        id="checkbox2"
        onChange={[Function]}
        type="checkbox"
      />
      <label
        htmlFor="checkbox2"
        onClick={[Function]}
      />
    </div>
    <i
      className="ui-icon ui-icon--size-20 fill-currentColor emotion-2 emotion-3"
      onClick={undefined}
      style={Object {}}
      title={undefined}
    >
      <svg
        aria-hidden="true"
        height={20}
        width={20}
      >
        <use
          xlinkHref="#keyboard_arrow_down"
        />
      </svg>
    </i>
    <div
      className="emotion-8 emotion-9"
    >
      <div
        style={
          Object {
            "display": "none",
          }
        }
      />
    </div>
  </div>
  <div
    className="ui-btn__react-success-message"
  />
</button>
`;
